### PR TITLE
8264360: Loop strip mining verification fails with "should be on the backedge"

### PR DESCRIPTION
**Pull request by rwestrel**
The assert checks that loads that are found as uses of a loop back
edge have the back edge as control input. In this case, because that
code is executed during final graph reshaping (CastPPs are in the
process of being eliminated and memory operations inherit their
control inputs as precedence edges), the edge between a load and the
back edge is a precedence edge. Relaxing the assert is all that is
required.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8264360](https://bugs.openjdk.java.net/browse/JDK-8264360): Loop strip mining verification fails with "should be on the backedge"

### Reviewers
 * [Vladimir Kozlov](https://openjdk.java.net/census#kvn) (@vnkozlov - **Reviewer**) ⚠️ Review applies to 24fa528bea8ae7704b8086fb4caa348e2694104c
 * [Tobias Hartmann](https://openjdk.java.net/census#thartmann) (@TobiHartmann - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/3245/head:pull/3245` \
`$ git checkout pull/3245`

Update a local copy of the PR: \
`$ git checkout pull/3245` \
`$ git pull https://git.openjdk.java.net/jdk pull/3245/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 3245`

View PR using the GUI difftool: \
`$ git pr show -t 3245`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/3245.diff">https://git.openjdk.java.net/jdk/pull/3245.diff</a>

</details>

### CODE REVIEW

**Review by vnkozlov — APPROVED** [review]

File: test/hotspot/jtreg/compiler/loopstripmining/TestLoadOnBackedgeWithPrec.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+class a {
+    int g = 20;
+    float h = 2;
+    long b = 6;
+}
+
+public class TestLoadOnBackedgeWithPrec {
+    int c ;
+    a[] i =   {
+        new a()};
```
Comment (TobiHartmann) [inline test/hotspot/jtreg/compiler/loopstripmining/TestLoadOnBackedgeWithPrec.java:43]: Whitespaces and newline should be removed.

File: test/hotspot/jtreg/compiler/loopstripmining/TestLoadOnBackedgeWithPrec.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    float j() {
+        a k = new a();
+        float l = 5     ;
```
Comment (TobiHartmann) [inline test/hotspot/jtreg/compiler/loopstripmining/TestLoadOnBackedgeWithPrec.java:46]: Whitespaces before `;` should be removed.

File: test/hotspot/jtreg/compiler/loopstripmining/TestLoadOnBackedgeWithPrec.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+            }
+            for (int f = 0; f < 9; ++f) {
+                new a();
+            }
+            {
+                a[] m = {
+                    new a(), new a(), new a(),
+                    new a(), new a(), new a(),
+                    new a(), new a(), new a()};
+                c = i[0].g   + k.g;
```
Comment (TobiHartmann) [inline test/hotspot/jtreg/compiler/loopstripmining/TestLoadOnBackedgeWithPrec.java:60]: Whitespaces before `+` should be removed.

**Review by TobiHartmann — APPROVED** [review]
The fix looks good to me but you might want to clean up the test a bit before pushing (I've added some comments).

File: test/hotspot/jtreg/compiler/loopstripmining/TestLoadOnBackedgeWithPrec.java (hunk 3 above)
Comment (rwestrel) [reply test/hotspot/jtreg/compiler/loopstripmining/TestLoadOnBackedgeWithPrec.java]: Thanks for reviewing this. I made the change you suggested.

**Review by TobiHartmann — APPROVED** [review]

**Comment by rwestrel** [comment]
@vnkozlov @TobiHartmann Thanks for the reviews

**Pull request closed without merging**